lib/praxis/api_definition.rb

```diff
@@ -41,35 +46,45 @@ def trait(name, &block)
       if self.traits.has_key? name
         raise Exceptions::InvalidTrait.new("Overwriting a previous trait with the same name (#{name})")
       end
-      self.traits[name] = block
+      self.traits[name] = Trait.new(&block)
     end
 
-    # Setting info to the nil version, means setting it for all versions (if they don't  override them)
-    def info( version=nil, &block)
-      i = @infos[version]
-      i.instance_eval(&block)
-      i
+    # Setting info to the nil version, means setting it for all versions (if they don't override them)
+    def info(version=nil, &block)
+      if version.nil?
+        if block_given?
+          @global_info.instance_eval(&block)
+        else
+          @global_info
+        end
+      else
+        i = @infos[version]
+        if block_given?
+          i.instance_eval(&block)
+        end
+        i
+      end
     end
-    
+
     def describe
-      global_info = @infos[nil].describe
       data = Hash.new do |hash, version|
         hash[version] = Hash.new
       end
       # Fill in the "info" portion
       @infos.each do |version,info|
-        next unless version
-        info_hash = global_info.merge(info.describe)
-        [:name, :title].each do |attr|
-          raise "Error: API Global information for version '#{version}' does not have '#{attr}' defined. " unless info_hash.key? attr
+        data[version][:info] = info.describe
+      end
+
```

Comment:
yes, I know it would be partial, but at least, I'd like to be able to get things like description and such out of if (which me, the app builder, could write in a global way...including links to other docs...etc). If we put it in "global" at least a "client/ API browser" can get to that info and display if necessary.